src/test/java/integration/DiscountPerCounterpartyControllerIT.java

```diff
@@ -0,0 +1,267 @@
+        DiscountPerCounterparty discountPerCounterparty = discountsPerCounterparty.get(0);
+        
+        JSONObject inputJson = testHelper.getJsonObjectFromFile("json/discount-per-counterparty.json");
+        inputJson.put("discountUuid", newDiscount.getUuid().toString());
+        inputJson.put("counterpartyUuid", discountPerCounterparty.getCounterparty().getUuid().toString());
+        
+        given().
+                queryParam("token", discountPerCounterparty.getCounterparty().getUser().getToken().toString()).
+                contentType(APPLICATION_JSON_VALUE).
+                body(inputJson.toString()).
+        when().
+                put("/discounts-per-counterparty/{uuid}", discountPerCounterparty.getUuid().toString()).
+        then().
+                body("discountUuid", equalTo(newDiscount.getUuid().toString())).
+                statusCode(SC_OK).
+        extract().
+                path("uuid");
+    
+        DiscountPerCounterparty updatedDiscountPerCounterparty = discountPerCounterpartyService
+                .getEntityByUuid(discountPerCounterparty.getUuid(), discountPerCounterparty.getCounterparty().getUser());
+        
+        assertEquals(updatedDiscountPerCounterparty.getDiscount().getUuid().toString(), newDiscount.getUuid().toString());
+    }
+    
+    @Test
+    public void updateDiscountPerCounterparty_invalidTime() throws Exception {
+        Discount newDiscount = testHelper.createDiscount();
+        DiscountPerCounterparty discountPerCounterparty = discountsPerCounterparty.get(0);
+        
+        JSONObject inputJson = testHelper.getJsonObjectFromFile("json/discount-per-counterparty.json");
+        inputJson.put("discountUuid", newDiscount.getUuid().toString());
+        inputJson.put("counterpartyUuid", discountPerCounterparty.getCounterparty().getUuid().toString());
+        inputJson.put("fromDate", "2016-06-01T18:25:43.511Z");
+        inputJson.put("toDate", "2016-09-01T18:25:43.511Z");
+        
+        given().
+                queryParam("token", discountPerCounterparty.getCounterparty().getUser().getToken().toString()).
+                contentType(APPLICATION_JSON_VALUE).
+                body(inputJson.toString()).
+        when().
+                put("/discounts-per-counterparty/{uuid}", discountPerCounterparty.getUuid().toString()).
+        then().
+                statusCode(SC_BAD_REQUEST);
+        
+        DiscountPerCounterparty existedDiscountPerCounterparty = discountPerCounterpartyService
+                .getEntityByUuid(discountPerCounterparty.getUuid(), discountPerCounterparty.getCounterparty().getUser());
+        
+        // make sure entity has not been updated.
+        assertEquals(
+                discountPerCounterparty.getDiscount().getUuid().toString(),
+                existedDiscountPerCounterparty.getDiscount().getUuid().toString());
+    }
+    
+    @Test
+    public void updateDiscountPerCounterparty_invalidTimeExpiredDiscount() throws Exception {
```

Comment:
I think we should delete it
Just one negative test to be sure if DiscountPerCounterparty is not in the range of the Discount